### ISSUE TITLE
WebGUI upload never completes

### ISSUE THREAD

**Issue by tsmr2015** [owncloud/core#19433]
### Steps to reproduce
1. upload any file, any size using web interface

### Expected behaviour
The file is uploaded, progress bar completes and folder is refreshed.

### Actual behaviour
When I upload a file, the transfer progress bar correctly reaches 100% however it never seems to complete. If I go to the "Admin" page or log out/in I get the JS warning about possible loss of data. Then I go back to the folder and the file was uploaded correctly.

### Server configuration
**Operating system:** Debian 3.2.68-1+deb7u3

**Web server:** Apache/2.2.22 (Debian) 

**Database:** MySQL

**PHP version:** 5.4.44-0+deb7u1

**ownCloud version:** ownCloud 8.1.3 (stable)

**Updated from an older ownCloud or fresh install:** fresh

**List of activated apps:**

Enabled:
 - activity: 2.0.2
 - calendar: 0.7.3
 - documents: 0.10.2
 - files: 1.1.10
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - firstrunwizard: 1.1
 - provisioning_api: 0.2
 - templateeditor: 0.1
 - updater: 0.6

Disabled:
 - contacts
 - encryption
 - external
 - files_external
 - files_locking
 - gallery
 - user_external
 - user_ldap
 - user_webdavauth

**The content of config/config.php:**

<?php
$CONFIG = array (
  'instanceid' => 'REDACTED',
  'passwordsalt' => 'REDACTED',
  'secret' => 'REDACTED',
  'trusted_domains' =>
  array (
    0 => 'REDACTED',
  ),
  'datadirectory' => '/var/owncloud/data',
  'overwrite.cli.url' => 'REDACTED',
  'dbtype' => 'mysql',
  'version' => '8.1.3.0',
  'dbname' => 'REDACTED',
  'dbhost' => 'REDACTED',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'REDACTED',
  'dbpassword' => 'REDACTED',
  'logtimezone' => 'UTC',
  'installed' => true,
  'loglevel' => '0'
);

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** none

### Client configuration
**Browser:** Waterfox 40.0.2 + Chrome 45.0.2454.101 m

**Operating system:** Win7

#### Web server error log

no errors in server log

#### ownCloud log (data/owncloud.log)
```
{"reqId":"ureveP7qRejsPC3Z3cM2","remoteAddr":"","app":"DeleteOrphanedSharesJob","message":"0 orphaned share(s) deleted","level":0,"time":"2015-09-29T08:30:01+00:00","method":"--","url":"--"}
{"reqId":"2B4HPNNKlBMoXkdVfeNG","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166055","level":1,"time":"2015-09-29T08:34:14+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=1"}
{"reqId":"2B4HPNNKlBMoXkdVfeNG","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166051","level":1,"time":"2015-09-29T08:34:14+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=1"}
{"reqId":"2B4HPNNKlBMoXkdVfeNG","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166049","level":1,"time":"2015-09-29T08:34:15+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=1"}
{"reqId":"RpT7Y27Seh5qRDJ8ONLJ","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166055","level":1,"time":"2015-09-29T08:35:21+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=1"}
{"reqId":"RpT7Y27Seh5qRDJ8ONLJ","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166051","level":1,"time":"2015-09-29T08:35:21+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=1"}
{"reqId":"RpT7Y27Seh5qRDJ8ONLJ","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166049","level":1,"time":"2015-09-29T08:35:21+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=1"}
{"reqId":"I2qoG+dfOHVIs9nVZRA7","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166055","level":1,"time":"2015-09-29T08:35:29+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=1&includeUpdateInfo=1"}
{"reqId":"I2qoG+dfOHVIs9nVZRA7","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166051","level":1,"time":"2015-09-29T08:35:29+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=1&includeUpdateInfo=1"}
{"reqId":"I2qoG+dfOHVIs9nVZRA7","remoteAddr":"REDACTED","app":"core","message":"No update found at the ownCloud appstore for app 166049","level":1,"time":"2015-09-29T08:35:29+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=1&includeUpdateInfo=1"}
{"reqId":"Xc2hGxOvSa0qAjDOQmsR","remoteAddr":"REDACTED","app":"caldav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 401 No basic authentication headers were found\",\"Code\":0,\"Trace\":\"#0 \\\/www\\\/REDACTED\\\/lib\\\/private\\\/connector\\\/sabre\\\/auth.php(149): Sabre\\\\DAV\\\\Auth\\\\Backend\\\\AbstractBasic->authenticate(Object(Sabre\\\\DAV\\\\Server), 'ownCloud')\\n
#1 \\\/www\\\/REDACTED\\\/lib\\\/private\\\/connector\\\/sabre\\\/auth.php(122): OC\\\\Connector\\\\Sabre\\\\Auth->auth(Object(Sabre\\\\DAV\\\\Server), 'ownCloud')\\n
#2 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php(118): OC\\\\Connector\\\\Sabre\\\\Auth->authenticate(Object(Sabre\\\\DAV\\\\Server), 'ownCloud')\\n
#3 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#4 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n
#5 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(456): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n
#6 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#7 \\\/www\\\/REDACTED\\\/apps\\\/calendar\\\/appinfo\\\/remote.php(57): Sabre\\\\DAV\\\\Server->exec()\\n
#8 \\\/www\\\/REDACTED\\\/remote.php(132): require_once('\\\/www\\\/REDACTED...')\\n
#9 {main}\",\"File\":\"\\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Backend\\\/AbstractBasic.php\",\"Line\":70}","level":0,"time":"2015-09-29T08:44:34+00:00","method":"OPTIONS","url":"\/remote.php\/caldav\/principals\/REDACTED\/"}
{"reqId":"oCGhC1lcNxRJn+XOLuMk","remoteAddr":"REDACTED","app":"caldav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 401 No basic authentication headers were found\",\"Code\":0,\"Trace\":\"#0 \\\/www\\\/REDACTED\\\/lib\\\/private\\\/connector\\\/sabre\\\/auth.php(149): Sabre\\\\DAV\\\\Auth\\\\Backend\\\\AbstractBasic->authenticate(Object(Sabre\\\\DAV\\\\Server), 'ownCloud')\\n
#1 \\\/www\\\/REDACTED\\\/lib\\\/private\\\/connector\\\/sabre\\\/auth.php(122): OC\\\\Connector\\\\Sabre\\\\Auth->auth(Object(Sabre\\\\DAV\\\\Server), 'ownCloud')\\n
#2 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php(118): OC\\\\Connector\\\\Sabre\\\\Auth->authenticate(Object(Sabre\\\\DAV\\\\Server), 'ownCloud')\\n
#3 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#4 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n
#5 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(456): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n
#6 \\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#7 \\\/www\\\/REDACTED\\\/apps\\\/calendar\\\/appinfo\\\/remote.php(57): Sabre\\\\DAV\\\\Server->exec()\\n
#8 \\\/www\\\/REDACTED\\\/remote.php(132): require_once('\\\/www\\\/REDACTED...')\\n
#9 {main}\",\"File\":\"\\\/www\\\/REDACTED\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Backend\\\/AbstractBasic.php\",\"Line\":70}","level":0,"time":"2015-09-29T08:44:35+00:00","method":"PROPFIND","url":"\/remote.php\/caldav\/calendars\/REDACTED\/"}
{"reqId":"opOyUUSJWFlKJscYmBfP","remoteAddr":"","app":"DeleteOrphanedSharesJob","message":"0 orphaned share(s) deleted","level":0,"time":"2015-09-29T08:45:02+00:00","method":"--","url":"--"}
```

**Comment by tsmr2015** [comment]
As reported on forums, the issue is resolved by disabling the "Share Files 0.6.2" app

**Comment by PVince81** [comment]
@tsmr2015 thanks for the detailed information.

Is the upload to a shared folder ? I guess it isn't since it works when disabling sharing.
When sharing is enabled, what options were set in the "Admin" page in the "Sharing" section ?
Does it happen for any user or is it a specific user ? Does that user receive shares from someone ?

**Comment by PVince81** [comment]
@tsmr2015 do you have the forum link, in case there is additional clues ?

**Comment by PVince81** [comment]
Also, does Webdav upload work ? Desktop or mobile client ? This could be an additional clue regarding the code paths.

**Comment by PVince81** [comment]
Okay, so bascially the file is uploaded properly but something is happening after writing the file and it returns an error instead. Unfortunately I don't see anything related in the log.

**Comment by tsmr2015** [comment]
@PVince81 All sharing features have been disabled via the admin panel
@PVince81 I'm guessing you already know that but the file is uploaded correctly (md5 hash is verified) only the transfer process in the web interface never completes. Also no issues whatsoever with mac sync client

**Comment by PVince81** [comment]
@tsmr2015 can you open the network console during the upload, and when it's done, check the response from ajax/upload.php ?

**Comment by tsmr2015** [comment]
I have now activated the sharing features and re-enabled the "Share Files 0.6.2" app, file transfer completes without any issues.
Having the app enabled while disabling sharing generates the problem.

![share_settings](https://cloud.githubusercontent.com/assets/14886333/10161019/6c13538e-66a1-11e5-9317-abd1a50ebb6b.PNG)

**Comment by tsmr2015** [comment]
[{"id":5692,"parentId":"4507","date":"September 29, 2015 at 12:03:09 PM GMT+2","mtime":1443520989000,"icon":"\/core\/img\/filetypes\/package-x-generic.svg","name":"REDACTED.rar","permissions":27,"mimetype":"application\/x-rar-compressed","size":158881,"type":"file","etag":"6bd2e6c2242072544ac3bfe220b6e35f","status":"success","originalname":"REDACTED.rar","uploadMaxFilesize":537919488,"maxHumanFilesize":"513 MB","directory":"\/stuff\/work\/REDACTED"}]

**Comment by PVince81** [comment]
Hmmm right, I can't find the ticket about it any more. It seems that disabling "Allow apps to use the Share API" can cause trouble when the "Share files" app is still enabled.

**Comment by PVince81** [comment]
Hmm, I tried the following:

1. Setup v8.1.3
1. Go to admin page
1. Disable "Allow apps to use the Share API"
1. Upload a file

Getting this JS error after upload:
```
Uncaught TypeError: Cannot read property 'loadIcons' of undefined(anonymous function) @ share.js?v=d028082f1d751c91934050a3efd2da32:75
x.event.dispatch @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
x.event.add.v.handle @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
x.event.trigger @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
e.event.trigger @ jquery-migrate.min.js?v=d028082f1d751c91934050a3efd2da32:2(anonymous function) @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
x.extend.each @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:4
x.fn.x.each @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:4
x.fn.extend.trigger @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
FileActions.display @ fileactions.js?v=d028082f1d751c91934050a3efd2da32:411
FileList._renderRow @ filelist.js?v=d028082f1d751c91934050a3efd2da32:948
FileList.add @ filelist.js?v=d028082f1d751c91934050a3efd2da32:865(anonymous function) @ filelist.js?v=d028082f1d751c91934050a3efd2da32:2067
x.event.dispatch @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
x.event.add.v.handle @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
x.event.trigger @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
e.event.trigger @ jquery-migrate.min.js?v=d028082f1d751c91934050a3efd2da32:2
(anonymous function) @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
x.extend.each @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:4
x.fn.x.each @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:4
x.fn.extend.trigger @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:5
$.Widget._trigger @ jquery-ui.custom.js?v=d028082f1d751c91934050a3efd2da32:787
$.widget._onDone @ jquery.fileupload.js?v=d028082f1d751c91934050a3efd2da32:794
(anonymous function) @ jquery-ui.custom.js?v=d028082f1d751c91934050a3efd2da32:405
(anonymous function) @ jquery.fileupload.js?v=d028082f1d751c91934050a3efd2da32:836
x.Callbacks.c @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:4
x.Callbacks.p.fireWith @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:4k @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:6
x.ajaxTransport.send.r @ jquery.min.js?v=d028082f1d751c91934050a3efd2da32:6
```

Let me have a look, hopefully a quick fix.

**Comment by PVince81** [comment]
Ah yes, this was fixed on master https://github.com/owncloud/core/pull/17343

**Comment by PVince81** [comment]
Backport PR is here, for 8.1: https://github.com/owncloud/core/pull/19439

**Comment by MorrisJobke** [comment]
Fixed with 8.1.4
